### ISSUE TITLE
PR #1789: DBNet training diverges (loss 0.21 -> 1.48)

### ISSUE THREAD

**Issue by ooples** [ooples/AiDotNet#1854]
## Summary
`DBNet` training does not reduce loss — loss rises during training. Reproduces in isolation (not a CI flake).

## Failing tests (shard: ModelFamily - Generated Layers A-F)
- `DBNetTests.Training_ShouldReduceLoss`
- `DBNetTests.TrainingError_ShouldNotExceedTestError`

## Error
```
Training did not reduce loss: initial=0.206629, final=1.484411.
Gradient computation or parameter update may be broken.
  at NeuralNetworkModelTestBase.Training_ShouldReduceLoss() line 486
```
Loss grows ~7x over training — gradient sign/scale problem or a severed tape (raw op instead of Engine.* breaking backprop), or a missing residual/normalization causing the optimizer to diverge. DBNet is a differentiable-binarization text-detection net; check the DB head and its loss wiring against the paper (Liao et al. 2020).

## Repro
```
dotnet test tests/AiDotNet.Tests/AiDotNetTests.csproj --framework net10.0 -c Release --no-build \
  --filter "FullyQualifiedName~DBNetTests.Training_ShouldReduceLoss|FullyQualifiedName~DBNetTests.TrainingError_ShouldNotExceedTestError"
```

## Classification
Real pre-existing model bug ([5-15 s] in isolation). Fix training/gradient path per paper; do not weaken the test.

**Comment by ooples** [comment]
## Diagnosis (3 root causes — the first two are prerequisites, the third is the blocker)

1. **Missing output sigmoid.** `CreateDefaultDBNetLayers` ends in a raw `ConvolutionalLayer(1,...)` (unbounded logits), but the default loss `BinaryCrossEntropyLoss` has `RequiresProbabilityInputs = true` and the model treats the head output as the probability map. Per Liao et al. 2020 §3.1, P = sigmoid(...). Add a final `ActivationLayer(SigmoidActivation)`.

2. **Double parameter update.** `Train()` calls `TrainWithTape(...)` (which runs the FULL forward→backward→optimizer step) AND THEN `CollectParameterGradients()` + `UpdateParameters()` (a second, plain-SGD update). Every step applies the gradient twice. Fix: keep only `TrainWithTape` (matches every other TrainWithTape-based model, e.g. AttentionNetwork).

3. **BLOCKER — unstable deep plain-conv backbone.** Even with (1)+(2), the model still diverges on its true BCE objective (measured BCE climbs 0.67 → 2.02 over training). `CreateDefaultDBNetLayers` is a ~8-layer conv stack + 6 BatchNorms with **no residual connections** (the comment literally says "ResNet blocks (simplified to conv layers)"). A deep plain-conv net with batch-1 BN is not stably trainable under Adam. Applying just (1)+(2) actually regresses `MoreData_ShouldNotDegrade` (currently green) because the sigmoid changes the dynamics the unstable backbone can't handle over 200 iters.

**Proper fix:** make the backbone paper-faithful ResNet-18 with real residual blocks (skip connections) so deep training is stable, THEN apply (1) + (2). This is a backbone rework, not a one-line change — hence left for a focused pass rather than shipping interacting partial fixes that regress a currently-green test.

Reverted my (1)+(2) exploration to keep the tree clean; documenting here so the rework starts from this diagnosis.

**Comment by ooples** [comment]
Verified fixed on PR #1789 — commit `7f9f8852d` (sigmoid DB head + stable batch-1 backbone) across `DBNet.cs`, `LayerHelper.cs` and `TestScaffoldGenerator.cs`. Now linked with `Closes #1854`.
